src/pharmpy/internals/df.py

```diff
@@ -1,6 +1,31 @@
+from __future__ import annotations
+
+from hashlib import sha256
+from typing import Iterator, Union
+
 from pharmpy.deps import pandas as pd
 
 
-def hash_df(df) -> int:
-    values = pd.util.hash_pandas_object(df, index=True).values
-    return hash(tuple(values))
+def _pd_hash_values(obj: Union[pd.Index, pd.Series, pd.DataFrame]) -> pd.Series:
+    # NOTE We explicit all arguments for future-proofing
+    return pd.util.hash_pandas_object(
+        obj, index=False, encoding='utf8', hash_key='0123456789123456', categorize=True
+    )
+
+
+def _df_hash_values(df: pd.DataFrame) -> Iterator[pd.Series]:
+    yield _pd_hash_values(df.columns)
+    yield _pd_hash_values(df.index)
+    yield _pd_hash_values(df.dtypes)
+    yield _pd_hash_values(df)
+
+
+def hash_df_runtime(df: pd.DataFrame) -> int:
```

Comment:
Is this used anywhere?

Reply:
No, but I think it could be used if we ever need to put dataframes in sets or dicts.